src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/math/RobustBrentSolver.java

```diff
@@ -0,0 +1,439 @@
+     * @param min lower bound for the root
+     * @param max upper bound for the root
+     * @param numBisections number of bisections for "combing" the search interval
+     * @param depth the fractal depth of the search grid
+     * @return the root
+     * @throws TooManyEvaluationsException if too many calls are made to the objective function
+     * @throws NoBracketingException if the root could not be ultimately bracketed
+     */
+    public double solve(final int maxEval,
+                        @Nonnull final UnivariateFunction objFunc,
+                        @Nullable final UnivariateFunction meritFunc,
+                        @Nullable final MeritPolicy meritPolicy,
+                        final double min, final double max,
+                        final int numBisections, final int depth)
+            throws TooManyEvaluationsException, NoBracketingException {
+        Utils.validateArg(meritFunc != null || meritPolicy != null, "Either a merit function or a merit policy" +
+                        " must be provided");
+        Utils.nonNull(objFunc, "The objective function must be non-null");
+        ParamUtils.isPositive(maxEval, "Maximum function evaluations must be positive");
+        ParamUtils.isPositiveOrZero(max - min, "The search interval is ill-posed");
+        ParamUtils.isPositiveOrZero(numBisections, "Number of search interval bisections must be non-negative");
+        ParamUtils.isPositiveOrZero(depth, "The search depth must be non-negative");
+
+        evaluations = 0;
+
+        /* create a search grid */
+        final double[] xSearchGrid = createFractalSearchGrid(min, max, numBisections, depth);
+
+        /* evaluate the function on the search grid */
+        final double[] fSearchGrid = new double[xSearchGrid.length];
+        for (int k = 0; k < xSearchGrid.length; k++) {
+            fSearchGrid[k] = objFunc.value(xSearchGrid[k]);
+            evaluations++;
+        }
+
+        final List<Bracket> bracketsList = detectBrackets(xSearchGrid, fSearchGrid);
+        if (bracketsList.isEmpty()) {
+            throw new NoBracketingException(min, max, fSearchGrid[0], fSearchGrid[fSearchGrid.length-1]);
+        } else {
+            final List<Double> roots = new ArrayList<>(bracketsList.size());
+            for (final Bracket bracket : bracketsList) {
+                final BrentSolverWithInitialFuncEvals solver = new BrentSolverWithInitialFuncEvals(relativeAccuracy,
+                        absoluteAccuracy, functionValueAccuracy);
+                try {
+                    final double x0 = 0.5 * (bracket.xMin + bracket.xMax);
+                    final double x = solver.solve(maxEval, objFunc, bracket.xMin, bracket.xMax, x0,
+                            bracket.fMin, bracket.fMax, objFunc.value(x0));
+                    roots.add(x);
+                } catch (final TooManyEvaluationsException | NoBracketingException ex) {
+                    /* nothing to do */
```

Comment:
We get `NoBracketingException` all the time -- most of the search intervals don't contain a root; that's why I ditch these intermediate `NoBracketingException`s.
If a root is found, it is guaranteed to be valid.